Lib/http/server.py

```diff
@@ -332,6 +333,11 @@ def parse_request(self):
                 return False
         self.command, self.path = command, path
 
+        # bpo-43223: The purpose of replacing '//' with '/' is to provide protection 
+        # against open redirect attack reside within http.server module which can be 
+        # triggered if the path contains '//' at the beginning.
+        self.path = re.sub(r'(/)+', '/', self.path)
```

Comment:
I would prefer to restrict effects of the security fix, and so stick to the beginning of the URL, only replace //path with /path:

```suggestion
        self.path = re.sub(r'^/+', '/', self.path)
```